[PATCH v4] validation: ipsec: improve initialization of unit tests

asasidharan opened the pull request:
Preiously when IPsec unit tests were initialized ODP queue
and ODP pktio were created by default however ODP queue is used
by async unit tests and pktio is used by inline unit tests. Sync
unit tests use neither ODP queue or ODP pktio. This patch updates
IPsec units test initialization so that only required elements are
created for a given mode of tests.

Signed-off-by: Lukasz Bartosik <lbartosik@marvell.com>
Signed-off-by: Anoob Joseph <anoobj@marvell.com>
Signed-off-by: Aakash Sasidharan <asasidharan@marvell.com>

psavol commented on test/validation/api/ipsec/ipsec.c:
```diff
@@ -1035,16 +1035,21 @@ int ipsec_init(odp_instance_t *inst)
+		if (ODP_PKTIO_INVALID == pktio) {
+			fprintf(stderr, "IPsec pktio creation failed.\n");
+			return -1;
+		}
```
ipsec_term() function seems to need modification as well. Otherwise, it prints "pktio not found" errors for sync/async test cases.


`	pktio = odp_pktio_lookup("loop");
	if (ODP_PKTIO_INVALID != pktio) {
		if (odp_pktio_close(pktio))
			fprintf(stderr, "IPsec pktio close failed.\n");
	} else {
		fprintf(stderr, "IPsec pktio not found.\n");
	}`

asasidharan replied:
> 
> 
> ipsec_term() function seems to need modification as well. Otherwise, it prints "pktio not found" errors for sync/async test cases.
> 
> ```
> pktio = odp_pktio_lookup("loop");
> if (ODP_PKTIO_INVALID != pktio) {
> 	if (odp_pktio_close(pktio))
> 		fprintf(stderr, "IPsec pktio close failed.\n");
> } else {
> 	fprintf(stderr, "IPsec pktio not found.\n");
> }
> ```

Will fix this in the next revision.

asasidharan replied:
Updated V2 with fix in ipsec_term. With the new change, there is no lookup performed for each resources. Instead, the handles available in the suite_context is used while destroying the resources.

asasidharan replied:
@psavol @JannePeltonen Does this change look okay now?

psavol commented on test/validation/api/ipsec/ipsec.c:
```diff
@@ -1092,32 +1097,23 @@ int ipsec_config(odp_instance_t ODP_UNUSED inst)
-	odp_pktio_t pktio;
+	odp_pool_t pool = suite_context.pool;
+	odp_queue_t out_queue = suite_context.queue;
+	odp_pktio_t pktio = suite_context.pktio;
```
It seems that depending on the test case some of these may be uninitilized. suite_context is static variable set to zero, but XXX_INVALID value is implementation defined (may not be zero). Maybe ipsec_init() should init all these handles to INVALIDs.

asasidharan replied:
Agreed. With V3, the suite_context fields pool, queue and pktio are initialized to corresponding XXX_INVALID values in ipsec_init().

psavol commented:
V3 is otherwise OK, but change-id must be removed from the commit log entry. Checkpatch warns about it as well. You can add my reviewed-by tag to v4 (Reviewed-by: Petri Savolainen <petri.savolainen@nokia.com>).

ERROR: Remove Gerrit Change-Id's before submitting upstream
#18: 
Change-Id: Id2cc453fad4b864d94cfcd87e5ce5fe94ca08cf2
total: 1 errors, 0 warnings, 0 checks, 137 lines checked

asasidharan commented:
> 
> 
> V3 is otherwise OK, but change-id must be removed from the commit log entry. Checkpatch warns about it as well. You can add my reviewed-by tag to v4:
> 
> `Reviewed-by: Petri Savolainen <petri.savolainen@nokia.com>`
> 
> ERROR: Remove Gerrit Change-Id's before submitting upstream
> #18:
> Change-Id: Id2cc453fad4b864d94cfcd87e5ce5fe94ca08cf2
> total: 1 errors, 0 warnings, 0 checks, 137 lines checked

Sorry. Missed to remove the Change-Id. Updated V4 with the Change-Id removed and added Reviewed-by tag.

anoobj commented:
@JannePeltonen @psavol Do you have any more comments on this?
